enhancements/storage/csi-driver-operator-merge.md

```diff
@@ -0,0 +1,321 @@
+* [IBM PowerVS](https://github.com/openshift/ibm-powervs-block-csi-driver-operator/)
+* [IBM VPC](https://github.com/openshift/ibm-vpc-block-csi-driver-operator/)
+* [OpenStack Cinder](https://github.com/openshift/openstack-cinder-csi-driver-operator/)
+* [OpenStack Manila](https://github.com/openshift/csi-driver-manila-operator/)
+* [oVirt Disk](https://github.com/openshift/ovirt-csi-driver-operator/) (*)
+* [Secret Store](https://github.com/openshift/secrets-store-csi-driver-operator/) (OLM)
+* [Shared Resource](https://github.com/openshift/csi-driver-shared-resource-operator/)
+* [VMware vSphere](https://github.com/openshift/vmware-vsphere-csi-driver-operator/)
+
+Operators marked by (*) will be removed soon, as OCP stops supporting the particular cloud provider.
+
+All these operators have their own separate github repository (i.e. 15 of them). All of them share the
+same [CSI driver operator library in library-go](https://github.com/openshift/library-go/tree/master/pkg/operator/csi),
+so their code is quite small and mostly just configure the shared library.
+
+* When a CVE is found in the shared library, it needs to be fixed in all 15 repositories. Similarly, Kubernetes go
+  packages (like k8s.io/client-go) needs to be bumped in each of them. This is a lot of work, and it's easy to miss one
+  of the repositories.
+
+* Recently we have started adding HyperShift support to these operators (AWS EBS first). This means that we need to add
+  a new functionality to all these operators and it's easier to do it in a single repository.
+
+### User Stories
+
+* As an OpenShift engineer, I want to fix a CVE in a vendored package in all CSI driver operators at once, so that I
+  don't have to fix it in their individual git repositories separately.
+  
+* As an OpenShift cluster admin, I don't see any difference in my cluster. All CSI drivers + their operators work as
+  before and use the same API.
+  
+* Explicitly, as OpenShift cluster admin, I am still able to install and uninstall AWS EBS and GCP filestore CSI drivers
+  from OLM as before.
+  
+### Goals
+
+* Simplify maintenance of CSI driver operators that are part of OCP and built + shipped by ART.
+
+### Non-Goals
+
+* Create a generic framework for CSI driver operators to be used by 3rd party vendors.
+
+## Proposal
+
+1. Merge all CSI driver operators listed above into a single repository, github.com/openshift/csi-operator, together
+   with the shared CSI driver operator library (i.e. move it from library-go to this repository). This will be a gradual
+   process over several OCP releases - we will move few operator at a time.
+1. Until all CSI driver operators are merged into the common repository, update them in their existing repos to vendor
+   the shared library from github.com/openshift/csi-operator instead of library-go, so we can fix bugs only on a single
+   place. Remove all CSI controllers from library-go, if possible.
```

Comment:
As an alternative we could just update the operators as needed.

Reply:
Actually, there is one benefit of keeping existing controllers in library-go, while copying them to csi-operator too: it would allow us to refactor the controllers in csi-operator, breaking any compatibility with the library-go ones. I do not plan to do such breaking changes right now, but maybe it would be nice to have such a possibility.

Reply:
Done, controllers stay in library-go for now.